sql/mozfun/norm/truncate_version/udf.sql

```diff
@@ -1,24 +1,25 @@
 CREATE OR REPLACE FUNCTION norm.truncate_version(os_version STRING, truncation_level STRING)
-RETURNS STRING AS (
+RETURNS NUMERIC AS (
   CASE
   WHEN
     truncation_level = "minor"
   THEN
-    REGEXP_EXTRACT(os_version, r"^([0-9]+[.]?[0-9]+).*")
+    CAST(REGEXP_EXTRACT(os_version, r"^([0-9]+[.]?[0-9]+).*") AS NUMERIC)
   WHEN
     truncation_level = "major"
   THEN
-    REGEXP_EXTRACT(os_version, r"^([0-9]+).*")
+    CAST(REGEXP_EXTRACT(os_version, r"^([0-9]+).*") AS NUMERIC)
   ELSE
     NULL
   END
 );
 
 -- Tests
 SELECT
-  assert.equals("16.1", norm.truncate_version("16.1.1", "minor")),
-  assert.equals("16", norm.truncate_version("16.1.1", "major")),
-  assert.equals("10", norm.truncate_version("10", "minor")),
-  assert.equals("5.1", norm.truncate_version("5.1.5-ubuntu-foobar", "minor")),
-  assert.equals(CAST(NULL AS STRING), norm.truncate_version("5.1.5-ubuntu-foobar", "patch")),
-  assert.equals(CAST(NULL AS STRING), norm.truncate_version("foo-bar", "minor"))
+  assert.equals(16.1, norm.truncate_version("16.1.1", "minor")),
+  assert.equals(16.03, norm.truncate_version("16.03.1", "minor")),
```

Comment:
This case actually worries me a bit. Do we see minor versions like "03" in the wild? I could imagine a minor version "2" without the zero padding that should sort before "03" logically, but the numeric value we return is `16.03`.

I think I am worrying too much, and this is probably fine.

Reply:
Ubuntu numbers things like "16.04" (YY.MM), so it's not unheard of. But I don't think they mix/match versioning like you suggest for your pathological case: the next version after "16.04" was "16.10" IIRC.